pkg/kubectl/cmd/scale.go

```diff
@@ -122,43 +121,35 @@ func RunScale(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []stri
 		return err
 	}
 
-	infos, err := r.Infos()
-	if err != nil {
-		return err
-	}
-	info := infos[0]
-	mapping := info.ResourceMapping()
-	scaler, err := f.Scaler(mapping)
-	if err != nil {
-		return err
-	}
+	counter := 0
+	err = r.Visit(func(info *resource.Info, err error) error {
+		if err != nil {
+			return err
+		}
 
-	resourceVersion := cmdutil.GetFlagString(cmd, "resource-version")
-	if len(resourceVersion) != 0 && len(infos) > 1 {
-		return fmt.Errorf("cannot use --resource-version with multiple resources")
```

Comment:
@deads2k I had removed the checks for the variables `resourceVersion` and for `currentSize` (below) because as we traverse each file using `Visit()` now, they're both being scoped to a single item, not multiple, and it is still allowing the precondition to be validated.

Do you still think these checks are still necessary given that the usage of `infos` is being superseded by using `Visit()`?